### PR TITLE
Build in OSX on Travis

### LINKED ISSUE #126: Support OSX in Travis
~Probably using Docker?~
- [x] Compile without errors.
- [x] Pass all tests.

### PR DESCRIPTION
**Pull request by serin-delaunay**
Resolves #126.

Changes:
* Run two Travis jobs: Linux and OSX.
* Move Linux Travis commands into script (`travis-ubuntu.sh`).
* Make script for OSX Travis commands (`travis-osx.sh`):
  * Installs SFML with Homebrew.
  * Builds with CMake, make, and clang.
  * Runs tests and builds examples.
* Choose the script to run based on platform.
* Rename `wangscape` library to `wangscape-main` (Fixes [this bug](https://www.reddit.com/r/proceduralgeneration/comments/6e7is5/procedural_tileset_nicener_wangscape_v01/dici774/)).
* Fix compilation errors detected by clang:
  * Missing `<string>` header.
  * Negative numbers implicitly cast to unsigned types.
  * `std::div` used instead of `div`.
  * Order of initialisation in `TestRequiringOptions`.
* Make `googletest` a submodule and update CMakeLists.txt to use it from a submodule rather than an installed package. This also applies to Linux builds, which means the `googletest` package can be removed from the dockerfile.
* Update `README.md` with:
  * New OSX build instructions.
  * Updated Ubuntu build instructions.
  * Note of pre-compiled Windows package.
  * Changes to dependencies.

### CODE REVIEW

**Comment by serin-delaunay** [comment]
The `brew install` command in `README.md` might need `cmake`, and possibly `clang`. Travis already has them installed, and I don't have a Mac to test the instructiins on.

File: .gitmodules [hunk 1]
```diff
@@ -10,3 +10,6 @@
 	path = lib/libnoise
 	url = https://github.com/qknight/libnoise.git
 	ignore = dirty
+[submodule "lib/googletest"]
+	path = lib/googletest
+	url = https://github.com/google/googletest
```
Comment (hryniuk) [inline .gitmodules]: 👍

Comment (hryniuk) [inline .gitmodules]: 👍

File: README.md [hunk 2]
```diff
@@ -165,7 +161,36 @@ make
...
+cmake .. -DCMAKE_MODULE_PATH=/usr/local/opt/sfml/share/SFML/cmake/Modules
+make
+```
+
+### Testing
+
+To run **unit tests** just execute:
+
+```shell
+./bin/WangscapeTest ../doc
```
Comment (hryniuk) [inline README.md]: We could ask **Angwe** from Reddit for a review of this building process :).

**Pull request merged**